webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueCompilationCache.scala

```diff
@@ -119,32 +140,66 @@ class TechniqueCompilationStatusService(
 
 /**
   * Technique compilation output needs to be saved in a cache (frequent reads, we don't want to get files on the FS every time).
-  * 
-  * This cache is a simple in-memory one which only saves errors, 
+  *
+  * This cache is a simple in-memory one which only saves errors,
   * so it saves only compilation output stderr message in case the compilation failed.
   * It notifies the lift actor on update of the compilation status.
-  * 
+  *
   * It is used mainly to get errors in the Rudder UI, and is updated when API requests to update techniques are made,
   * when technique library is reloaded
   */
 class TechniqueCompilationErrorsCache(
-    actor: SimpleActor[UpdateCompilationStatus],
-    ref:   Ref[Map[(BundleName, Version), EditorTechniqueError]]
+    actor:     SimpleActor[UpdateCompilationStatus],
+    reader:    ReadTechniqueCompilationStatusService,
+    errorBase: Ref[Map[(BundleName, Version), EditorTechniqueError]]
 ) extends WriteTechniqueCompilationStatusService {
 
-  override def update(technique: EditorTechnique, output: TechniqueCompilationOutput): UIO[CompilationStatus] = {
-    val key = (technique.id, technique.version)
-    (if (output.isError) {
-       ref.updateAndGet(_ + (key -> EditorTechniqueError(technique.id, technique.version, technique.name, output.stderr)))
-     } else {
-       ref.updateAndGet(_ - key)
-     })
-      .map(m => {
-        val status = getStatus(m.values)
-        // update every time even if value is the same for now, actor will handle the change
-        actor ! UpdateCompilationStatus(status)
-        status
-      })
+  /*
+   * Update the internal cache and build a Compilation status
+   */
+  private[ncf] def updateStatus(results: List[EditorTechniqueResult]): UIO[CompilationStatus] = {
+    errorBase.updateAndGet { m =>
+      results.foldLeft(m) {
+        case (current, EditorTechniqueResult(id, version, name, CompilationResult.Error(error))) =>
+          current + ((id, version) -> EditorTechniqueError(id, version, name, error))
+        case (current, EditorTechniqueResult(id, version, _, CompilationResult.Success))         =>
+          current - ((id, version))
+      }
+    }.map(m => getStatus(m.values)).tap {
+      case CompilationStatusAllSuccess => ConfigurationStatusLoggerPure.info("All techniques have success compilation result")
+      case e: CompilationStatusErrors =>
+        val techniques = e.techniquesInError.map(t => s"${t.id.value}(v${t.version.value})").toList.mkString(",")
+        ConfigurationStatusLoggerPure.warn(
+          s"Found ${e.techniquesInError.size} techniques with compilation errors when starting server : ${techniques}"
+        )
+    }
```

Comment:
I think the place for logs would be in the more "batch" operation `getUpdateAndSync`, especially if techniques were to be updated one by one by the API (in that case the `when starting server` needs to be removed...)